common/src/main/java/com/wynntils/services/mapdata/type/MapArea.java

```diff
@@ -5,11 +5,17 @@
 package com.wynntils.services.mapdata.type;
 
 import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.type.BoundingBox;
 import java.util.List;
 
 public interface MapArea extends MapFeature {
     // The area is described by a polygon. This list is the sequence of
     // vertices of that polygon, ordered in a counterclockwise orientation.
     // The last segment of the polygon connects from the last vertice to the first.
     List<Location> getPolygonArea();
+
+    @Override
+    default boolean isVisible(BoundingBox boundingBox) {
```

Comment:
Just a note, this is technically not correct. An area could have all its points outside the bounding box, but it is still visibile inside the bounding box. In fact, this problem might be tricky to solve correctly. Or at least, I'm not a good enough game developer to know the answer straight ahead. Might be better to leave this unimplemented than pushing an incorrect implementation.